src/createServer.js

```diff
@@ -1,3 +1,68 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    const url = new URL(req.url, `http://${req.headers.host}`);
+    const originalText = url.pathname.slice(1);
+    const targetCase = url.searchParams.get('toCase');
+    const errors = [];
+
+    res.setHeader('Content-Type', 'application/json');
+
+    if (!originalText) {
+      errors.push({
+        message:
+          // eslint-disable-next-line max-len
+          'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!targetCase) {
+      errors.push({
+        message:
+          // eslint-disable-next-line max-len
+          '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    } else if (
+      !['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'].includes(targetCase)
+    ) {
+      errors.push({
+        message:
+          // eslint-disable-next-line max-len
+          'This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+      });
+    }
+
+    if (errors.length) {
+      res.statusCode = 400;
+      res.statusMessage = 'Bad request';
+      res.end(JSON.stringify({ errors }));
+
+      return;
+    }
+
+    const { originalCase, convertedText } = convertToCase(
+      originalText,
+      targetCase,
+    );
```

Comment:
This is a good use of destructuring to make the code cleaner and more readable. However, it's worth noting that if the `convertToCase` function throws an error, it will crash the server. Consider adding error handling here to ensure the server can recover gracefully from errors.